bundler/lib/bundler/compact_index_client.rb

```diff
@@ -21,24 +21,17 @@ class Error < StandardError; end
 
     require_relative "compact_index_client/cache"
     require_relative "compact_index_client/cache_file"
+    require_relative "compact_index_client/parser"
     require_relative "compact_index_client/updater"
 
-    attr_reader :directory
-
-    def initialize(directory, fetcher)
-      @directory = Pathname.new(directory)
-      @updater = Updater.new(fetcher)
-      @cache = Cache.new(@directory)
-      @endpoints = Set.new
-      @info_checksums_by_name = {}
-      @parsed_checksums = false
-      @mutex = Thread::Mutex.new
+    def initialize(directory, fetcher = nil)
+      @cache = Cache.new(directory, fetcher)
+      @parser = Parser.new(@cache)
     end
 
     def execution_mode=(block)
       Bundler::CompactIndexClient.debug { "execution_mode=" }
-      @endpoints = Set.new
-
+      @cache&.reset!
```

Comment:
Is `@cache` ever nil?

Reply:
oh! leftover from when this interacted with `@updater` which could be nil. I'll fix now that the `@cache` abstracts this situation.